hal/tests/TESTS/pin_names/pinvalidate.py

```diff
@@ -14,29 +14,62 @@
 limitations under the License.
 """
 
-import os
-import re
 import argparse
-import pathlib
 import json
+import os
+import pathlib
+import re
+import sys
 from tabulate import tabulate
+from enum import Enum
+
+
+class ReturnCode(Enum):
+    """Return codes."""
+
+    SUCCESS = 0
+    ERROR = 1
+    INVALID_OPTIONS = 2
+
+
+class TestCaseError(Exception):
+    """An exception for test case failure."""
+
+
+class ArgumentParserWithDefaultHelp(argparse.ArgumentParser):
+    """Subclass that always shows the help message on invalid arguments."""
+
+    def error(self, message):
+        """Error handler."""
+        sys.stderr.write("error: {}\n".format(message))
+        self.print_help()
+        raise SystemExit(ReturnCode.INVALID_OPTIONS.value)
+
+
+def find_target(target_name=""):
+    """Find a target."""
+    mbed_os_root = os.path.abspath(
+        os.path.join(
+            pathlib.Path(__file__).parent.absolute(), "..", "..", "..", ".."
+        )
+    )
```

Comment:
I think this would be preferable, at least for the exact way in which you are doing this (hard-coding the number of ancestors until the root):

```python
mbed_os_root = pathlib.Path(__file__).absolute().parent[5]
```

If I am not mistaken, taking the absolute path of an ancestor of an absolute path is redundant.

Reply:
I will let @gpsimenos decide how to find the root. Btw, there is no `.mbed` at the root, it's applications that have it.

Reply:
`cwd` returns the working directory, which is not necessarily the directory of the python file.

Reply:
You're right. It returns the working directory of where the interpreter is called. My mistake.

I was looking to avoid the `__file__` dunders, mainly because I tried it from the interpreter REPL and it failed. Of course, the REPL doesn't define a `__file__`, so it makes sense. Furthermore, if I wanted to use it this way: `import path.to.file.pinvalidate` that will still work.

In that case, still avoiding the `os` module:
```python
mbed_os_root = pathlib.Path(__name__).absolute().parents[5]
```
I think this should work. Also, putting it in a separate function as I suggest below would handle this twice.

There are some comments that `Path.resolve()` should do this too. I read a little, and I am not convinced, but it could be the best.

Also note: my previous example accidentally left off the `s` in `parents`.

Reply:
Done.